daffodil-test/src/test/scala/org/apache/daffodil/section07/variables/TestVariables.scala

```diff
@@ -59,6 +59,16 @@ class TestVariables {
   @Test def test_varInstance_09(): Unit = { runner.runOneTest("varInstance_09") }
   @Test def test_varInstance_10(): Unit = { runner.runOneTest("varInstance_10") }
   @Test def test_varInstance_11(): Unit = { runner.runOneTest("varInstance_11") }
+  @Test def test_varInstance_12(): Unit = { runner.runOneTest("varInstance_12") }
+
+  @Test def test_defineVariable_nonConstantExpression(): Unit = { runner.runOneTest("defineVariable_nonConstantExpression") }
+  // The following tests don't seem to play nicely with TDMLRunner and the errors are not detected/checked against
+  //@Test def test_circular_defineVariable_err(): Unit = { runner.runOneTest("circular_defineVariable_err") }
```

Comment:
Is there still a reason for these tests to be commented out? If so please create JIRA tickets for these.

Reply:
I meant to leave a comment about those tests in the pull request, but must have forgotten.

test_circular_defineVariable_err produces the error as expected with the given test case, but for some reason TDMLRunner fails to recognize it and the test just fails and prints the stack trace.  I'm guessing that it may be due to the fact that the test fails after schema compilation but before any infoset is generated.  Same issue applies for test_defineVariable_ref_noDefault_err.  I'm open to creating a ticket for it, but wanted to get opinions if I am simply doing something wrong in the test or if this sort of test should be moved to a CLI test instead (which shouldn't have any problem detecting the error).

test_defineVariable_ref_infoset_err runs into issues in the expression path handling code where it assumes that there will always be an infoset.  I will create a ticket for this bug.

Reply:
I think the bug that is failing and producing a stack trace that's got to be a TDML-runner issue. Do you have a negative test that defines two variables that are circular. That test should be able to expect an SDE and get it. If that's getting masked somehow, it's a TDML runner issue, or perhaps the error is being thrown of wrong type somehow. Wants to be a compile time SDE.

Reply:
Both the circular definition error and the test with an expression referencing a variable without a default value produce runtime schema definition errors, but they are not picked up by TDMLRunner for some reason.  I'll create a bug for the issue and point to the tests that trigger the issue.